tests/runner/kube_testplatform.go

```diff
@@ -296,9 +297,16 @@ func (c *KubeTestPlatform) SetAppEnv(name, key, value string) error {
 		return err
 	}
 
-	_, err := appManager.WaitUntilDeploymentState(appManager.IsDeploymentDone)
+	// 120s is the observed time to spawn a new test app POD and terminate the old one.
+	// This sleep time is important for the stream log to work.
+	time.Sleep(120 * time.Second)
```

Comment:
This seems extremely high, and we're already cutting close the 10 minutes limit for some of our test suites. The line below this one should gate this appropriately. If deployment state is running, your logs should stream properly. If you are seeing errors, it might be because the StreamContainerLogs() attempts to run against all pods in a deployment, and might try against a terminated pod, I think? The new pod should still work, even in the presence of this error.

Reply:
This one is needed because it is the time I observed for the new POD to be created and the old one to be terminated. Otherwise, tests can fail because the requests can go to any of the PODs and add flakiness.

Reply:
So WaitUntilDeploymentState doesn't sufficiently gate on the deployment being completed? Could we just update WaitUntilDeploymentState to also gate on the number of replicas being what we expect?

Reply:
I checked the code. It does that already. Let me remove this. Nice catch.